interpreter/cling/include/cling/Interpreter/Transaction.h

```diff
@@ -135,10 +136,19 @@ namespace cling {
     ///
     Transaction* m_Parent;
 
+    ///\brief RAII owning this Transaction.
+    ///
+    const PushTransactionRAII* m_Scope = nullptr;
+
     unsigned m_State : 3;
 
     unsigned m_IssuedDiags : 2;
 
+    ///\brief the Transaction is currently being unloaded. Currently,
+    /// used for ensuring system consistency when unloading transactions.
+    ///
+    unsigned m_Unloading : 1;
```

Comment:
```suggestion
    bool m_Unloading : 1;
```
more accurate or not legal?

Reply:
Legal, but wasting space: my bit is attached to the previous three bits; your `bool` needs a whole byte. You can take the address of a bool, you cannot take the address of my bitfield.

Reply:
>  your bool needs a whole byte.

hummm ... I missed in the standard (as copied by cppreferences) where it specifies that:
```
unsigned a : 1;
bool b : 1;
```
takes 12 bytes rather than 4 (as opposed to be "implementation dependent")

Reply:
And I missed your `: 1` after `bool`! :-) Will fix (in https://github.com/root-project/root/pull/7773 ).